Function/BookFunc/Services/BookingService.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Graph;
+using Microsoft.Identity.Client;
+using Microsoft.Graph.Auth;
+using TeamsMeetingBookFunc.Models;
+using TeamsMeetingBookFunc.Helpers;
+using Microsoft.Azure.WebJobs;
+
+namespace TeamsMeetingBookFunc.Services
+{
+    class BookingService
+    {
+        internal IConfigurationRoot Configuration { get; private set; }
+
+        internal GraphServiceClient GraphServiceClient { get; private set; }
+
+        #region Lazy and thread-safe singleton
+        private static readonly Lazy<BookingService> _current = new Lazy<BookingService>(() => new BookingService());
+
+        internal static BookingService Current => _current.Value;
+
+        private BookingService()
+        {
+
+        }
```

Comment:
Move the initialization in the CTOR, so we are sure it will rune once only thanks to the Lazy<T>

Reply:
```suggestion
        private BookingService()
        {
              Configuration = BuildConfig();

            Uri authority = new Uri($"https://login.microsoftonline.com/{Configuration.GetConnectionStringOrSetting(ConfigConstants.TenantIdCfg)}");

            var app = PublicClientApplicationBuilder.Create(Configuration.GetConnectionStringOrSetting(ConfigConstants.ClientIdCfg))
                .WithAuthority(authority)
                .Build();

            GraphServiceClient = new GraphServiceClient(new UsernamePasswordProvider(app));
        }
```